### PR TITLE
telegram update handlers

### PR DESCRIPTION
**Pull request by frosthamster**

### CODE REVIEW

File: tg/client.go [hunk 1]
```diff
@@ -4,18 +4,22 @@ import (
...
+	chatID   tele.ChatID
+	chat     *tele.Chat
+	handlers []BotUpdateHandler
 }
 
 func NewClient(token string, chatID tele.ChatID) (*Client, error) {
 	bot, err := tele.NewBot(tele.Settings{
-		Token: token,
+		Token:  token,
+		Poller: &tele.LongPoller{Timeout: 10 * time.Second},
```
Comment (cauchy2384) [inline tg/client.go:22]: Wdyt about moving it to the config?

File: tg/client.go [hunk 2]
```diff
@@ -30,6 +34,26 @@ func NewClient(token string, chatID tele.ChatID) (*Client, error) {
 	}, nil
 }
 
+type BotUpdateHandler interface {
+	Match(c tele.Context) bool
+	Handle(c tele.Context) error
+}
+
+func (c *Client) RegisterHandler(handler BotUpdateHandler) {
+	c.handlers = append(c.handlers, handler)
```
Comment (cauchy2384) [inline tg/client.go:43]: Is it important to ensure that this method is called when tg bot is not started yet?

File: tg/client.go [hunk 3]
```diff
@@ -30,6 +34,26 @@ func NewClient(token string, chatID tele.ChatID) (*Client, error) {
...
+func (c *Client) Start() {
+	c.bot.Handle(tele.OnText, func(ctx tele.Context) error {
+		for _, handler := range c.handlers {
+			if handler.Match(ctx) {
+				handler.Handle(ctx)
+			}
+		}
+		return nil
+	})
+	go c.bot.Start()
```
Comment (cauchy2384) [inline tg/client.go:54]: I suspect there must be a Stop() call somewhere :)

File: tg/client.go [hunk 4]
```diff
@@ -30,6 +34,26 @@ func NewClient(token string, chatID tele.ChatID) (*Client, error) {
...
+}
+
+func (c *Client) RegisterHandler(handler BotUpdateHandler) {
+	c.handlers = append(c.handlers, handler)
+}
+func (c *Client) Start() {
+	c.bot.Handle(tele.OnText, func(ctx tele.Context) error {
+		for _, handler := range c.handlers {
+			if handler.Match(ctx) {
+				handler.Handle(ctx)
```
Comment (cauchy2384) [inline tg/client.go:49]: I expect an error to be handled.

File: boardwhite/handlers.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+func (r ReactionHandler) Match(c tele.Context) bool {
+	return true
+}
+
+// TODO: rewrite this to do something useful
+func (r ReactionHandler) Handle(c tele.Context) error {
+	user := c.Sender()
+	message := c.Message()
+
+	slog.Info("log", slog.Any("user", user.Username))
```
[the remark this answers is not in the document]
Comment (nburtsev) [reply boardwhite/handlers.go:21]: It's a test handler, to play around with reactions, I will rewrite it before we merge

File: tg/client.go (hunk 2 above)
Comment (nburtsev) [reply tg/client.go:44]: I believe so, any suggestions on how to achieve that?

**Pull request merged**